program/src/processor/new_order.rs

```diff
@@ -366,6 +385,27 @@ pub(crate) fn process(
         ],
     )?;
 
+    if let Some(a) = accounts.fee_referral_account {
+        let referral_fee_transfer_instruction = spl_token::instruction::transfer(
+            accounts.spl_token_program.key,
+            accounts.user_token_account.key,
+            a.key,
+            accounts.user_owner.key,
+            &[],
+            referral_fee,
+        )?;
+
+        invoke(
```

Comment:
This is going to error if the referrer does not have an account initialized. Might be better to do `.ok()`?